host/switchless_manager.c

```diff
@@ -0,0 +1,126 @@
+/* Copyright (c) Microsoft Corporation.  All rights reserved.
+ * Licensed under the MIT License. */
+
+#include "switchless_manager.h"
+#include <openenclave/internal/calls.h>
+#include "hostthread.h"
+#include "sgx/enclave.h"
+
+static oe_thread_return_t _enclave_worker_thread(oe_thread_arg_t arg)
+{
+    oe_enclave_t* enclave = (oe_enclave_t*)arg;
+
+    oe_ecall(
+        enclave,
+        OE_ECALL_LAUNCH_ENCLAVE_WORKER,
+        (uint64_t)(&enclave->switchless_manager.switchless),
+        NULL);
+
+    return OE_THREAD_RETURN_VAL;
+} /* _enclave_worker_thread */
+
+void oe_switchless_manager_init(oe_enclave_t* enclave)
+{
+    enclave->switchless_manager.switchless.state = OE_SWITCHLESS_STATE_STOPPED;
+    oe_lockless_queue_init(
+        &(enclave->switchless_manager.switchless.ecall_queue));
+} /* oe_switchless_manager_init */
+
+oe_result_t oe_switchless_manager_startup(oe_enclave_t* enclave)
+{
+    oe_result_t result = OE_OK;
```

Comment:
The pattern is to initialize result with OE_FAILURE or OE_UNEXPECTED and then to set it to OE_OK only when all the operations have succeeded.

Reply:
changed.

Reply:
oe_switchless_manager_startup now returns OE_OK in all cases unless thread create fails.